### PR TITLE
Add werk.technica werk.one keyboard

### PR DESCRIPTION
**Pull request by ebastler**
## Description
This PR adds the WT One keyboard to QMK

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/werk_technica/one/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                   KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, NK_TOGG, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,          KC_TRNS,          KC_TRNS,          KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS,  KC_TRNS
+        ),
+
+};
+
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
```
Comment (zvecr) [inline keyboards/werk_technica/one/keymaps/default/keymap.c:50]:
```suggestion
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
```

File: keyboards/werk_technica/one/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,                   KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, NK_TOGG, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS,          KC_TRNS,          KC_TRNS,          KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS,  KC_TRNS
+        ),
+
+};
+
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
```
Comment (zvecr) [inline keyboards/werk_technica/one/keymaps/via/keymap.c:50]:
```suggestion
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
```

File: keyboards/werk_technica/one/werk_one.c
Comment (zvecr) [inline keyboards/werk_technica/one/werk_one.c:1]: This file can be removed, as its the default behaviour <https://github.com/qmk/qmk_firmware/blob/master/quantum/encoder.c#L90>.

File: keyboards/werk_technica/one/info.json [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+    "diode_direction": "COL2ROW",
+    "features": {
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true,
+        "encoder": true,
+        "lto": true
```
Comment (zvecr) [inline keyboards/werk_technica/one/info.json:23]:
```suggestion
        "encoder": true
```

File: keyboards/werk_technica/one/info.json [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+    "bootloader": "stm32-dfu",
+    "processor": "STM32F411",
+    "board": "GENERIC_STM32_F411XE",
+    "usb": {
+        "pid": "0x0001",
+        "vid": "0x7774",
+        "device_version": "1.0.0"
+    },
+    "diode_direction": "COL2ROW",
+    "features": {
```
Comment (zvecr) [inline keyboards/werk_technica/one/info.json:15]:
```suggestion
    "build": {
        "lto": true
    },
    "features": {
```

File: keyboards/werk_technica/one/info.json [hunk 5]
```diff
@@ -0,0 +1,315 @@
+{
+    "manufacturer": "werk_technica",
+    "keyboard_name": "one",
+    "maintainer": "ebastler",
+    "url": "https://werktechnica.com/",
+    "bootloader": "stm32-dfu",
+    "processor": "STM32F411",
+    "board": "GENERIC_STM32_F411XE",
```
Comment (zvecr) [inline keyboards/werk_technica/one/info.json:8]:
```suggestion
```

File: keyboards/werk_technica/one/info.json [hunk 6]
```diff
@@ -0,0 +1,315 @@
...
+    "matrix_pins": {
+        "cols": [ "A6", "A7", "B0", "B1", "B10", "B12", "B13", "B14", "B15", "A8", "A10", "A13", "A5", "A4", "C15" ],
+        "rows": [ "A0", "A1", "A2", "A3", "B9", "B8" ]
+    },
+    "encoder": {
+        "rotary": [
+            { 
+                "pin_a": "B5", 
+                "pin_b": "A15",
+                "resolution": 4
```
Comment (zvecr) [inline keyboards/werk_technica/one/info.json:34]:
```suggestion
                "pin_b": "A15"
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/werk_technica/one/werk_one.c
Comment (ebastler) [reply keyboards/werk_technica/one/werk_one.c:1]: Oh, I was not aware that a fallback from encoder map exists. Thank you!

I also just realized it still had an outdated file name from when the board folder was werk_one and therefore not used for compiling anyway. File removed.

File: keyboards/werk_technica/one/info.json
Comment (fauxpark) [inline keyboards/werk_technica/one/info.json:1]: Please run `qmk format-json -i` on this file.

**Review by fauxpark — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by ebastler** [comment]
Sorry for being pushy, but we're one month in with no suggestions for changes or anything else, and I kinda need this merged before the boards ship to customers... Is there any news on this PR?

File: keyboards/werk_technica/one/keymaps/via/rules.mk [hunk 7]
```diff
@@ -0,0 +1,2 @@
+VIA_ENABLE=yes
```
Comment (nooges) [inline keyboards/werk_technica/one/keymaps/via/rules.mk:1]:
```suggestion
VIA_ENABLE = yes 
```

**Pull request merged**